torch/fx/_graph_pickler.py

```diff
@@ -0,0 +1,568 @@
+import dataclasses
+import importlib
+import io
+import pickle
+from abc import abstractmethod
+from typing import Any, Callable, Dict, NewType, Optional, Tuple, Type, TypeVar, Union
+from typing_extensions import override, Self
+
+import torch
+import torch.utils._pytree as pytree
+from torch._guards import TracingContext
+from torch._subclasses.fake_tensor import FakeTensor, FakeTensorMode, Tensor
+from torch._subclasses.meta_utils import (
+    MetaConverter,
+    MetaTensorDesc,
+    MetaTensorDescriber,
+)
+from torch.fx.experimental.sym_node import SymNode
+from torch.fx.experimental.symbolic_shapes import ShapeEnv
+from torch.utils._mode_utils import no_dispatch
+
+
+_SymNodeT = TypeVar("_SymNodeT", torch.SymInt, torch.SymFloat)
+
+
+class GraphPickler(pickle.Pickler):
+    """
+    GraphPickler is a Pickler which helps pickling fx graph - in particular
+    GraphModule.
+    """
+
+    def __init__(self, file: io.BytesIO) -> None:
+        super().__init__(file, protocol=pickle.HIGHEST_PROTOCOL)
```

Comment:
This makes it so that upgrades to pickle/python version break BC, right? Is that okay? Is there a more stable protocol choice?

Reply:
My original thinking was that the pickling was ephemeral so highest would be best, but actually there's no reason someone couldn't insert a cache and make it longer-lasting.